### DISCUSSION TITLE
dde-shell  交互缺失——在已启动的icon上滑动滚轮不再展示所有窗口

### DISCUSSION THREAD

**Discussion by shenmo7192** [linuxdeepin/developer-center#8871] [Features Request & Ideas | 特性请求 & 头脑风暴]
在deepin 20.9上，对dock上的图标滑动滚轮可以展示所有窗口
dde-shell上无法交互
配合这个bug会导致非常酸爽的工作效率下降

https://bbs.deepin.org.cn/post/273265

**Comment by tsic404** [comment]
现在程序的身份识别是[`dde-application-manager`在做](https://github.com/linuxdeepin/dde-shell/blob/master/panels/dock/taskmanager/desktopfileamparser.cpp#L147-L172), AM 识别是根据`cgroup`信息来的而不是根据窗口的实际身份。
就会导致`a`拉起启动`b`，导致`b`被识别成`a`。因为`b`是`a`的子进程，继承了`a`的`cgroup`信息。目前在做的`workaround`就是修改`a`拉起`b`的方式，去call am 上对应的 dbus 接口，而不是直接运行。或者是`b`被唤醒时，可以挪到正确的`cgroup`下面。

**Comment by tsic404** [comment]
dde-shell 滚轮滚动相关目前应该是没有做，后面应该要补上。任务管理模块目前缺的功能还比较多，只实现了基本的能力，一些交互比如拖拽一些受支持的文件类型到程序上可以自动打开等都暂未实现。
